Extending/Packages/Package-Actions/custom-package-actions.md

```diff
@@ -0,0 +1,120 @@
+---
+versionFrom: 8.0.0
+---
+
+# Creating custom package actions
+
+In addition to utilizing the [built-in package actions](index.md), you can also create your own package actions. Package actions are custom code that runs on install and uninstall of a package. You can do whatever you want in a package action - however some things are more common than others.
+
+## Basic package action implementation
+
+To create a custom package action you need to create a new class and implement the `IPackageAction` interface. The empty package action would then look like this:
+
+```cs
+using System.Xml.Linq;
+using Umbraco.Core.PackageActions;
+
+namespace CustomPackageAction.PackageActions
+{
+    public class CustomPackageAction : IPackageAction
+    {
+        public string Alias() => "CustomPackageAction";
+
+        public bool Execute(string packageName, XElement xmlData)
+        {
+            // do things on install
+            return true;
+        }
+
+        public bool Undo(string packageName, XElement xmlData)
+        {
+            // if you want to clean up things added in the Execute method do it here
+            return true;
+        }
+    }
+}
+```
+
+A package action consists of 3 mandatory methods:
+
+**Alias()** 
+In this method you specify the package action alias. This is used when you select a package action in the backoffice. It can look like this:
```

Comment:
"This is used when you select a package action in the backoffice."

_how about changing this to something like_

"This is used in the XML definition of the package actions that you specify when creating a package in the backoffice."